impeller/renderer/backend/vulkan/render_pass_vk.cc

```diff
@@ -126,6 +128,9 @@ SharedHandleVK<vk::RenderPass> RenderPassVK::CreateVKRenderPass(
     );
     TextureVK::Cast(*stencil->texture).SetLayout(barrier);
   }
+  if (old_renderpass != nullptr) {
+    return old_renderpass;
+  }
```

Comment:
If we hit this clause, was all the work above here for?  It looks like it's creating a builder and barrier, then never using it.

Reply:
We still need to perform the layout transitions on all attachments. I'm not sure if we want to try to re-arrange that to minimize the amount of builder code executed, which is itself pretty cheap.

Reply:
I'd like to keep this code as is, I don't agree that it is less confusing. Partially because I suspect we may eventually need the render pass descriptor to check compatibility, and partially because I think it will be more difficult to make sure those parts stay "in sync".